packages/core/src/messagingProtocols/proteus/ProteusService/ProteusService.ts

```diff
@@ -173,4 +178,82 @@ export class ProteusService {
       state: response.errored ? PayloadBundleState.CANCELLED : PayloadBundleState.OUTGOING_SENT,
     };
   }
+
+  private async encryptPayloadForSession(
+    sessionId: string,
+    plainText: Uint8Array,
+  ): Promise<SessionPayloadBundle | undefined> {
+    this.logger.log(`Encrypting payload for session ID "${sessionId}"`);
+
+    let encryptedPayload: Uint8Array;
+
+    try {
+      const payloadAsArrayBuffer = await this.coreCryptoClient.proteusEncrypt(sessionId, plainText);
+      encryptedPayload = new Uint8Array(payloadAsArrayBuffer);
+    } catch (error) {
+      const notFoundErrorCode = 2;
+      if ((error as any).code === notFoundErrorCode) {
+        // If the session is not in the database, we just return undefined. Later on there will be a mismatch and the session will be created
+        return undefined;
+      }
+      this.logger.error(`Could not encrypt payload: ${(error as Error).message}`);
+      encryptedPayload = new Uint8Array(Buffer.from('💣', 'utf-8'));
+    }
+
+    return {encryptedPayload, sessionId};
+  }
```

Comment:
actually we should try to use `coreCrypto.proteusEncryptBatched` this should simplify things a bit and should be way more performant (as we would avoid many back and forth between js and wasm)

Reply:
Cool, didn't know that, will refactor the code then :D